nncf/quantization/passes.py

```diff
@@ -34,8 +37,10 @@ def transform_to_inference_graph(
         that also can be interpreted as inputs (ReadValue).
     :return: NNCFGraph in the inference style.
     """
-    inference_nncf_graph = remove_shapeof_subgraphs(nncf_graph, shapeof_metatypes, read_variable_metatypes)
-    inference_nncf_graph = filter_constant_nodes(nncf_graph, read_variable_metatypes)
+    inference_nncf_graph = deepcopy(nncf_graph)
+    inference_nncf_graph = remove_shapeof_subgraphs(inference_nncf_graph, shapeof_metatypes, read_variable_metatypes)
+    inference_nncf_graph = filter_constant_nodes(inference_nncf_graph, read_variable_metatypes)
+    inference_nncf_graph = insert_noops_instead_constants(inference_nncf_graph, nncf_graph, metatypes_to_insert_noop)
```

Comment:
If we do `insert_noops_instead_constants` first and then apply `filter_constant_nodes` then `insert_noops_instead_constants` can have a more scalable signature. At least you will not need the original `nncf_graph`